addons/pos_sms/data/sms_data.xml

```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" ?>
+<odoo>
+    <data noupdate="1">
+        <record id="sms_template_data_point_of_sale" model="sms.template">
+            <field name="name">POS: Sent Order Confirmation via Text</field>
+            <field name="model_id" ref="point_of_sale.model_pos_order"/>
+            <field name="body">
+                {{ (object.company_id.name + ': Your Order ' + ("{:05d}".format(int(object.session_id.id))+'-'+"{:03d}".format(int(object.config_id.id))+'-'+"{:04d}".format(int(object.sequence_number)))+' was processed succesfully with amount ' + object.currency_id.symbol + ' ' + str(object.amount_total) + '. Use Order '+("{:05d}".format(int(object.session_id.id)) +'-' +"{:03d}".format(int(object.config_id.id))+'-'+"{:04d}".format(int(object.sequence_number)))+' for further reference')}}
```

Comment:
this is way to complicated.
In addition, you want to show the user the actual reference, not recompute what you think the  reference would be. The fact that the string `Order` appears in the reference is not a problem. 
> Your order with reference: Order 12345 was processed ...

 is fine.
You don't have to remove the string `Order` from the reference. You can just leave it.
Also, try to not reinvent formatting for the monetary value. Maybe you can use `formatLang` ( `from odoo.tools.misc import formatLang ` )

Reply:
try to use `obj.currency_id.format`

Reply:
it's normal to have: 
> Here is your invoice INV/2024/00027

( basically repeating "invoice" - "INV" ) 
We do not strip the "INV" part of the reference, we do not say "Here is your INV/2024....", we say "here is your invoice/order/quotation,etc: {{inv_reference}}"
reference:
![image](https://github.com/odoo/odoo/assets/118446179/9d31298d-8f26-4ae1-80c8-42d672a1b15d)